### DISCUSSION TITLE
Switching `node` to `control-plane` role.

### DISCUSSION THREAD

**Discussion by mprochowski** [k3s-io/k3s#9238] [Q&A] [answered]
Hi,
I have cluster (5xrpi - ubuntu 22.04) with 1 `control-plane` and 4 `node`, which I want to covert to multi `control-plane` environment. I was converted DB from sqlite to etcd, but I don't know how i should switch `node` to `control-plane`.

I'm thinking about two solutions:
1. Safely remove `node` from cluster (drain, cordon, etc.) and create new `control-plane` instance of it. - This is not preferred.
2. Somehow switch existing `node` instance to `control-plane` - maybe I could add `server` option to systemd start command?

What is the "right" scenario or what you suggest to switch `node` to `control-plane`?

**Comment by brandond** [comment, marked as answer]
Yes, all thats necessary from a K3s perspective is to change the startup command from `k3s agent` to `k3s server`. If you are comfortable renaming the systemd unit file, and modifying the contents, that is the simplest way to do this.

**Comment by mprochowski** [comment]
@brandond Thanks a lot, everything works perfectly.
